src/test/statistics/chunk_statistics/range_filter_test.cpp

```diff
@@ -196,98 +162,119 @@ TYPED_TEST(RangeFilterTest, CanPruneOnBounds) {
 
   // SQL's between is inclusive
-  EXPECT_FALSE(filter->can_prune(PredicateCondition::Between, {103}, {123456}));
+  EXPECT_FALSE(filter->can_prune(PredicateCondition::Between, TypeParam{103}, TypeParam{123456}));
+
+  // TODO(bensk1): as soon as non-inclusive between predicates are implemented, testing
+  // a non-inclusive between with the bounds exactly on the value bounds would be humongous:
+  //  EXPECT_TRUE(filter->can_prune(PredicateCondition::BetweenNONINCLUSIVE, TypeParam{103}, TypeParam{123456}));
 }
 
 // Test larger value ranges.
 TYPED_TEST(RangeFilterTest, LargeValueRange) {
-  std::random_device rd;
-  auto rng = std::mt19937(rd());
-
-  // values on which is the range filter is later built on
-  pmr_vector<TypeParam> values;
-  values.reserve(1'000);
-
-  // Multiplying by 0.6 guarantees that the range filter will always yield a gap range
-  // right in the middle around value zero. Note, does not apply for unsigned values.
-  const TypeParam lower_bound_low_range = static_cast<TypeParam>(0.4 * std::numeric_limits<TypeParam>::lowest());
-  const TypeParam upper_bound_low_range = static_cast<TypeParam>(0.6 * lower_bound_low_range);
-  const TypeParam upper_bound_high_range = static_cast<TypeParam>(0.4 * std::numeric_limits<TypeParam>::max());
-  const TypeParam lower_bound_high_range = static_cast<TypeParam>(0.6 * upper_bound_high_range);
-
-  // We randomly create values between min_value(TypeParam) to 0.6*min_value(TypeParam) and
-  // 0.6*max(TypeParam) to max(TypeParam). Any value in between should be prunable.
-  distribution<TypeParam> lower_range_distribution(lower_bound_low_range, upper_bound_low_range);
-  distribution<TypeParam> high_range_distribution(lower_bound_high_range, upper_bound_high_range);
-  for (auto i = size_t{0}; i < 1'000; ++i) {
-    values.push_back(get_random_number<TypeParam>(rng, lower_range_distribution));
-    values.push_back(get_random_number<TypeParam>(rng, high_range_distribution));
-  }
+  const pmr_vector<TypeParam> values{static_cast<TypeParam>(0.4 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.38 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.36 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.30 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.28 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.36 * std::numeric_limits<TypeParam>::max()),
+                                     static_cast<TypeParam>(0.38 * std::numeric_limits<TypeParam>::max()),
+                                     static_cast<TypeParam>(0.4 * std::numeric_limits<TypeParam>::max())};
+
+  // A filter with 3 ranges, has two gaps: (i) 0.28*lowest-0.36*max and (ii) 0.36*lowest-0.30*lowest
```

Comment:
can you just add EXPECTs for this? I feel like that would already do 80% of the testing you're trying to achieve here.